code/modules/mob/living/carbon/carbon.dm

```diff
@@ -1135,85 +1135,128 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, list(/obj/machinery/atmospherics/unary/ven
-	else
-		visible_message("<span class='warning'>[user] cannot force anymore of [toEat] down [src]'s throat.</span>")
-		return 0
-	if(!toEat.instant_application)
-		if(!do_mob(user, src))
-			return 0
-	forceFedAttackLog(toEat, user)
-	visible_message("<span class='warning'>[user] forces [src] to [toEat.apply_method] [toEat].</span>")
-	return 1
+/mob/living/carbon/proc/forceFed(obj/item/reagent_containers/to_eat, mob/user, fullness)
+	if(fullness > (600 * (1 + overeatduration / 1000)))
+		visible_message("<span class='warning'>[user] cannot force anymore of [to_eat] down [src]'s throat.</span>")
+		return FALSE
 
-/mob/living/carbon/proc/forceFedAttackLog(obj/item/reagent_containers/food/toEat, mob/user)
-	add_attack_logs(user, src, "Fed [toEat]. Reagents: [toEat.reagents.log_list(toEat)]", toEat.reagents.harmless_helper() ? ATKLOG_ALMOSTALL : null)
+	visible_message("<span class='warning'>[user] attempts to force [src] to swallow [to_eat].</span>")
+	if(!do_after(user, 3 SECONDS, TRUE, src))
+		return FALSE
+	forceFedAttackLog(to_eat, user)
+	visible_message("<span class='warning'>[user] forces [src] to swallow [to_eat].</span>")
+	return TRUE
 
+/mob/living/carbon/proc/forceFedAttackLog(obj/item/reagent_containers/to_eat, mob/user)
+	add_attack_logs(user, src, "Fed [to_eat]. Reagents: [to_eat.reagents.log_list(to_eat)]", to_eat.reagents.harmless_helper() ? ATKLOG_ALMOSTALL : null)
 
 /*TO DO - If/when stomach organs are introduced, override this at the human level sending the item to the stomach
 so that different stomachs can handle things in different ways VB*/
-/mob/living/carbon/proc/consume(obj/item/reagent_containers/food/toEat, bitesize_override, can_taste_container = TRUE)
-	var/this_bite = bitesize_override ? bitesize_override : toEat.bitesize
-	if(!toEat.reagents)
+/mob/living/carbon/proc/consume(obj/item/reagent_containers/food/to_eat, bitesize_override)
+	var/this_bite = bitesize_override ? bitesize_override : to_eat.bitesize
+	if(!to_eat.reagents)
 		return
 	if(satiety > -200)
-		satiety -= toEat.junkiness
-	if(toEat.consume_sound)
-		playsound(loc, toEat.consume_sound, rand(10,50), 1)
-	if(toEat.reagents.total_volume)
-		if(can_taste_container)
-			taste(toEat.reagents)
-		var/fraction = min(this_bite/toEat.reagents.total_volume, 1)
+		satiety -= to_eat.junkiness
+	if(to_eat.consume_sound)
+		playsound(loc, to_eat.consume_sound, rand(10, 50), TRUE)
+	if(to_eat.reagents.total_volume)
+		taste(to_eat.reagents)
+		var/fraction = min(this_bite / to_eat.reagents.total_volume, 1)
 		if(fraction)
-			toEat.reagents.reaction(src, toEat.apply_type, fraction)
-			toEat.reagents.trans_to(src, this_bite*toEat.transfer_efficiency)
+			to_eat.reagents.reaction(src, REAGENT_INGEST, fraction)
+			to_eat.reagents.trans_to(src, this_bite)
+
+/mob/living/carbon/proc/consume_patch_or_pill(obj/item/reagent_containers/p_or_p, user) // p_or_p = patch or pill
```

Comment:
i don't like `p_or_p`, maybe `medicine` as the var name instead? if you have to add a comment explaining the name its probably not a good name lol